pkg/util/labels/labels_test.go

```diff
@@ -148,11 +149,47 @@ func TestCloneSelectorAndAddLabel(t *testing.T) {
 				"foo4": "12",
 			},
 		},
+		{
+			labels:     labels,
+			labelKey:   "foo4",
+			labelValue: "89",
+			matchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      "tier",
+					Operator: "In",
+					Values:   []string{"cache"},
+				},
+			},
+			want: map[string]string{
+				"foo1": "bar1",
+				"foo2": "bar2",
+				"foo3": "bar3",
+				"foo4": "89",
+			},
+		},
+		{
+			labels:     labels,
+			labelKey:   "foo4",
+			labelValue: "89",
+			matchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      "tier",
+					Operator: "In",
+					Values:   nil,
+				},
+			},
+			want: map[string]string{
+				"foo1": "bar1",
+				"foo2": "bar2",
+				"foo3": "bar3",
+				"foo4": "89",
+			},
+		},
 	}
 
 	for _, tc := range cases {
-		ls_in := metav1.LabelSelector{MatchLabels: tc.labels}
-		ls_out := metav1.LabelSelector{MatchLabels: tc.want}
+		ls_in := metav1.LabelSelector{MatchLabels: tc.labels, MatchExpressions: tc.matchExpressions}
+		ls_out := metav1.LabelSelector{MatchLabels: tc.want, MatchExpressions: tc.matchExpressions}
 
 		got := CloneSelectorAndAddLabel(&ls_in, tc.labelKey, tc.labelValue)
 		if !reflect.DeepEqual(got, &ls_out) {
```

Comment:
It's not right here
```
t.Errorf("got %v, want %v", got, tc.want)
```
tc.want is only part of `CloneSelectorAndAddLabel`'s result.